From bcdaef952166d11121f19bfb1b913444447e7b13 Mon Sep 17 00:00:00 2001
From: Ronan Le Martret <ronan.lemartret@iot.bzh>
Date: Thu, 26 Nov 2020 14:03:29 +0000
Subject: [PATCH 06/11] m3ulcb.dts: add definition for Cortex-R7 remoteproc

Add Mailbox and remoteproc, also add the shm area for rpmsg.

Note that this patch disable the support of mfis-lock and mfis-as,
since register area are conflicting.

Signed-off-by: Ronan Le Martret <ronan.lemartret@iot.bzh>
---
 .../arm64/boot/dts/renesas/r8a77960-ulcb.dts | 28 ++++++++++++++
 arch/arm64/boot/dts/renesas/r8a77960.dtsi      | 38 ++++++++++++-------
 2 files changed, 53 insertions(+), 13 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a77960-ulcb.dts b/arch/arm64/boot/dts/renesas/r8a77960-ulcb.dts
index 09b70f35852a..18b3ef23487d 100644
--- a/arch/arm64/boot/dts/renesas/r8a77960-ulcb.dts
+++ b/arch/arm64/boot/dts/renesas/r8a77960-ulcb.dts
@@ -35,6 +35,26 @@
 		#size-cells = <2>;
 		ranges;
 
+		cr7_ram: cr7_ram@0x40040000 {
+			no-map;
+			reg = <0x0 0x40040000 0x0 0x1fc0000>;
+		};
+
+		vdev0vring0: vdev0vring0@42000000 {
+			no-map;
+			reg = <0x0 0x42000000 0x0 0x1000>;
+		};
+
+		vdev0vring1: vdev0vring1@42010000 {
+			no-map;
+			reg = <0x0 0x42010000 0x0 0x1000>;
+		};
+
+		vdev0buffer: vdev0buffer@42020000 {
+			no-map;
+			reg = <0x0 0x42020000 0x0 0x4000>;
+		};
+
 		/* device specific region for Lossy Decompression */
 		lossy_decompress: linux,lossy_decompress@54000000 {
 			no-map;
@@ -111,3 +131,11 @@
 &vspi0 {
 	status = "okay";
 };
+
+&cr7_rproc {
+	memory-region = <&cr7_ram>, <&vdev0vring0>, <&vdev0vring1>, <&vdev0buffer>;
+	mboxes = <&mfis 0>, <&mfis 1>;
+	mbox-names = "tx", "rx";
+	rcar,syscfg-rsc-tbl = <&mfisarembr 0x0 0xFFFFFFFF>;
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/renesas/r8a77960.dtsi b/arch/arm64/boot/dts/renesas/r8a77960.dtsi
index ae9902d35ea0..b06388c6cd1f 100755
--- a/arch/arm64/boot/dts/renesas/r8a77960.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a77960.dtsi
@@ -811,19 +811,11 @@
 		#size-cells = <2>;
 		ranges;
 
-		mfis: mfis@e6260000 {
-			compatible = "renesas,mfis-lock-r8a7796",
-				     "renesas,mfis-lock";
-			reg = <0 0xe6260000 0 0x1000>;
-			#hwlock-cells = <1>;
-
-			mfis_as: mfis-as {
-				compatible = "renesas,mfis-as-r8a7796",
-					     "renesas,mfis-as";
-				interrupts = <GIC_SPI 180 IRQ_TYPE_LEVEL_HIGH>;
-				interrupt-names = "eicr0";
-				renesas,mfis-ch = <0>;
-			};
+		cr7_rproc: cr7@40040000 {
+			compatible = "renesas,rcar-cr7";
+			power-domains = <&sysc R8A7796_PD_CR7>;
+			resets = <&cpg 222>;
+			status = "disabled";
 		};
 
 		rwdt: watchdog@e6020000 {
@@ -1088,6 +1080,26 @@
 			power-domains = <&sysc R8A7796_PD_ALWAYS_ON>;
 		};
 
+		mfis: mailbox@e6260400 {
+			compatible = "renesas,rcar-ipcc";
+			#mbox-cells = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0 0xe6260400 0 0x10>;
+			interrupts =
+				<GIC_SPI 224 IRQ_TYPE_LEVEL_HIGH>,
+				<GIC_SPI 225 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "tx", "rx";
+			clocks = <&cpg  CPG_MOD 213>;
+			power-domains = <&sysc R8A7796_PD_ALWAYS_ON>;
+			resets = <&cpg 213>;
+		};
+
+		mfisarembr: mfisarembr@e6260460{
+			compatible = "simple-bus", "syscon", "simple-mfd";
+			reg = <0 0xe6260460 0 0x20>;
+		};
+
 		i2c0: i2c@e6500000 {
 			#address-cells = <1>;
 			#size-cells = <0>;
-- 
2.31.1

